app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,72 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign = nil)
+    @assign = {
+      :new     => assign[:new],
+      :current => assign[:current]
+    }
+
+    build = false unless @assign[:new][:assign_to]
```

Comment:
Well, no other tree sets `build` to false - and `build=false` doesn't quite prevent the tree from getting rendered I think.

But maybe you already solved it by that `return nil if alert_profile_get_assign_to_objects_empty?` and there's still that `- if @assign[:obj_tree]` in the partial, so hopefully it should not be needed.